### PR TITLE
Run trimming tests as AOT tests

### PR DESCRIPTION
**Pull request by MichalStrehovsky**
Not everything is passing, so I baselined this. Some we'll probably exclude permanently, others are more concerning and we need to determine if it's test issues or product issues.

Cc @dotnet/ilc-contrib

### CODE REVIEW

File: src/libraries/tests.proj [hunk 1]
```diff
@@ -654,8 +654,20 @@
...
+    <DisabledNativeAotTestAppProjects Include="$(MSBuildThisFileDirectory)\System.Net.Http\tests\TrimmingTests\System.Net.Http.TrimmingTests.proj" />
+    <DisabledNativeAotTestAppProjects Include="$(MSBuildThisFileDirectory)\System.Reflection.DispatchProxy\tests\TrimmingTests\System.Reflection.DispatchProxy.TrimmingTests.proj" />
+    <DisabledNativeAotTestAppProjects Include="$(MSBuildThisFileDirectory)\System.Linq.Expressions\tests\TrimmingTests\System.Linq.Expressions.TrimmingTests.proj" />
+    <DisabledNativeAotTestAppProjects Include="$(MSBuildThisFileDirectory)\System.Private.Xml\tests\TrimmingTests\System.Private.Xml.TrimmingTests.proj" />
+    <DisabledNativeAotTestAppProjects Include="$(MSBuildThisFileDirectory)\System.Runtime\tests\System.Runtime.Tests\TrimmingTests\System.Runtime.TrimmingTests.proj" />
+    <DisabledNativeAotTestAppProjects Include="$(MSBuildThisFileDirectory)\System.Text.Json\tests\System.Text.Json.Tests\TrimmingTests\System.Text.Json.TrimmingTests.proj" />
+
+    <NativeAotTestAppProjects Include="$(MSBuildThisFileDirectory)*\tests\**\*.NativeAotTests.proj;$(MSBuildThisFileDirectory)*\tests\**\*.TrimmingTests.proj"
```
Comment (eerhardt) [inline src/libraries/tests.proj:669]: Should we just get rid of the `NativeAotTests.proj` extension (there is only 1 currently in the repo) and instead just have `TrimmingTests.proj`, which are run as both PublishTrimmed and PublishAot?

File: eng/testing/linker/trimmingTests.targets [hunk 2]
```diff
@@ -142,7 +142,7 @@
 
     <MSBuild Projects="@(TestConsoleApps)"
              Targets="Publish"
-             Properties="Configuration=$(Configuration);BuildProjectReferences=false;TargetOS=$(TargetOS);TargetArchitecture=$(TargetArchitecture)" />
+             Properties="Configuration=$(Configuration);BuildProjectReferences=false;TargetOS=$(TargetOS);TargetArchitecture=$(TargetArchitecture);_IsPublishing=true" />
```
Comment (eerhardt) [inline eng/testing/linker/trimmingTests.targets:145]: Why is this change necessary?

Comment (MichalStrehovsky) [reply eng/testing/linker/trimmingTests.targets:145]: The tests weren't building because they couldn't find AppHost. This is running the Publish target directly, which is very problematic because it bypasses logic in the SDK. We have been having a conversation with the SDK team for months. The fix is to do what real publish does and that's setting _IsPublishing. We do that in multiple places in this repo even though it's used/defined by the SDK.

File: src/libraries/tests.proj (hunk 1 above)
Comment (MichalStrehovsky) [reply src/libraries/tests.proj:669]: If you don't have concerns about running the NativeAotTests with trimming as well, that sounds good to me too.

Comment (eerhardt) [reply src/libraries/tests.proj:669]: This is the only NativeAotTests today:

https://github.com/dotnet/runtime/blob/4760cf03a6f6599c8f268004e8286a25d89f00b0/src/libraries/System.Diagnostics.DiagnosticSource/tests/NativeAotTests/DiagnosticSourceEventSourceTests.cs#L45-L84

It should work just fine with PublishTrimmed. It was written to test the MakeGenericType deep inside of DiagnosticSourceEventSource works with NativeAOT.

I think it would be best to just have 1 "kind of test" and it runs for both PublishTrimmed and PublishAot. It will make it easier to write these, and it will add a bit of coverage to both scenarios.

File: eng/testing/linker/trimmingTests.targets [hunk 3]
```diff
@@ -81,14 +81,17 @@
       <_additionalPropertiesString>@(_propertiesAsItems->'&lt;%(Identity)&gt;%(Value)&lt;/%(Identity)&gt;', '%0a    ')</_additionalPropertiesString>
     </PropertyGroup>
 
+    <!-- RunNativeAotTestApps trumps TestTrimming, same as PublishAot trumps PublishTrimmed -->
```
Comment (eerhardt) [inline eng/testing/linker/trimmingTests.targets:84]: (nit) this comment doesn't align with the code. The comment says one trumps the other. The code says if they are both true it is an error.

**Review by eerhardt — APPROVED** [review]
LGTM. Thank you for making this happen.
